crud/common/sharding/sharding_key.lua

```diff
@@ -106,12 +109,18 @@ function sharding_key_module.construct_as_index_obj_cache(metadata_map)
                                                                    metadata.space_format,
                                                                    metadata.sharding_key_def)
             if err ~= nil then
-                return err
+                if specified_space_name == space_name then
+                    result_err = err
+                else
+                    log.warn(err)
+                end
```

Comment:
We can issue a log entry in the former case as well (`log.warn()` for the latter case and `log.error()` for the former one).

A person who handles an error from the library and a person who looking into production logs are different ones usually.

Reply:
Done.